gemd/entity/base_entity.py

```diff
@@ -1,6 +1,9 @@
 """Base class for all entities."""
+from typing import Optional
+
 from gemd.entity.dict_serializable import DictSerializable
 from gemd.entity.case_insensitive_dict import CaseInsensitiveDict
+from gemd.entity.link_by_uid import LinkByUID
```

Comment:
This unfortunately opens the door for circular dependencies, but I had a pickle of a time getting `-> 'LinkByUID'` to pass flake8.